### DISCUSSION TITLE
4d nifti files cannot load successfully

### DISCUSSION THREAD

**Discussion by windyjunfeng** [Project-MONAI/MONAI#7188] [Q&A]
Hello, I meet some tough problems when loading 4D nifti files. I use the UNETR network. It seems that it can support the 4D nifti file because in_channels can be set bigger than 1 according to the codes' comment section. But when I set in_channels as 10, the error occurs. I try to solve the problem, but it still doesn't work. Could you please provide some help? Thank you very much.
The 4D nifti file size is (256,256,23,10).
The training command is as follows:
![image](https://github.com/Project-MONAI/MONAI/assets/58356387/0d7d4d3c-fbce-4632-b613-20ccb53771c2)
The specific error is as follows:
![image](https://github.com/Project-MONAI/MONAI/assets/58356387/71639588-7340-4e20-a21a-b1db41b98c4b)

**Comment by KumoLiu** [comment]
Hi @windyjunfeng, If 10 represents the channel dim, perhaps you should move it to the front and remove `EnsureChannelFirst` to prevent adding another channel dim.
https://github.com/Project-MONAI/MONAI/wiki/MONAI-Data-Shape-Assumption
Thanks!

**Comment by windyjunfeng** [comment]
@KumoLiu , thank you very much! Your solution works! However, I meet another problem when computing loss. The one hot function in networks/utils.py doesn't support multichannel in the label file. Could you please give some suggestions? Thanks a lot!
![image](https://github.com/Project-MONAI/MONAI/assets/58356387/c4ff929b-1302-4070-accb-5e49ee98863d)
![image](https://github.com/Project-MONAI/MONAI/assets/58356387/1b6f7d68-91a0-41d7-b8ec-32f6316c2b99)

**Comment by KumoLiu** [comment]
Hi @windyjunfeng, perhaps try argmax before sending to one-hot.
https://github.com/Project-MONAI/MONAI/blob/c3f9914b1fa4bb2e18f534563453cd3970c7e2e1/monai/transforms/post/array.py#L142

**Reply by windyjunfeng** [reply]
Thanks a lot for your help, although I don't use the AsDiscrete class lastly. I change the format of the label and integrate multi-channels into one single channel.

**Reply by windyjunfeng** [reply]
@KumoLiu Hello, sorry to trouble you again. I find one single channel cannot cope with the multi-label problem. Now the label file size is (5,256,256,23). 5 means five channels. Each voxel has a single label in each channel. But if five channels are integrated into one channel, the multi-label problem occurs. I try to solve it according to some discussion about it. I add the following code to the transforms.Compose.
`transforms.AsDiscreted(keys=["label"],to_onehot=True,num_classes=6)`
But errors still occur. Could you please provide some suggestions?
![image](https://github.com/Project-MONAI/MONAI/assets/58356387/57f6e496-3f3d-452e-9b98-40b31dd34bc2)
![image](https://github.com/Project-MONAI/MONAI/assets/58356387/36380f00-805b-418d-a0bc-2926f0fb1bc2)
![image](https://github.com/Project-MONAI/MONAI/assets/58356387/a4c6ea39-4bd8-402b-8c45-4d56152c0443)

**Reply by KumoLiu** [reply]
Hi @windyjunfeng, if you want to use `to_onehot`, you need to make sure that there is one channel with a length of 1, I found that your label has 5 channels.

**Reply by windyjunfeng** [reply]
Yes, I see. But if I integrate five channels into one single channel, some label masks will overlap due to the multi-label problem. Any brilliant suggestions? @KumoLiu Thank you~
